### DISCUSSION TITLE
Is there a way to determine whether the server was shut down using a handle or a SIGINT?

### DISCUSSION THREAD

**Discussion by megahomyak** [actix/actix-web#2800] [Q&A] [answered]
In my application, I am throwing the server handle to a separate thread before `.await`ing the server. When I shut the server down using the handle (to restart it on a different port or with a different hostname), I want it to live, and when it is killed using `Ctrl+C`, I want to shut down the entire program, so it would be useful to know if there is a simple way to determine the way the server was shut down. If not, I'll just share a boolean.

**Comment by robjtede** [comment, marked as answer]
No, the signal generated from ctrl-c just sends a stop signal like you would do manually. For your use case, it _could_ be easier using `HttpServer::disable_signals` and handling the ctrl-c manually, too.
